<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Annotation d’un génome</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Génotypage du bar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Bioinfo
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="tools-used.html">Outils utilisés</a>
    </li>
    <li>
      <a href="tools_description.html">Description des outils</a>
    </li>
    <li>
      <a href="tuto.html">Tutoriaux suivis</a>
    </li>
    <li>
      <a href="Troubleshooting.html">Troubleshooting</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Manip'
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="LaboSemaine1.html">Semaine 1</a>
    </li>
    <li>
      <a href="LaboSemaine2.html">Semaine 2</a>
    </li>
    <li>
      <a href="LaboSemaine3.html">Semaine 3</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Réunions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="premiereReunion_2702.html">Réunion du 27/02/2018</a>
    </li>
    <li>
      <a href="deuxiemeReunion_1403.html">Réunion du 14/03/2018</a>
    </li>
    <li>
      <a href="troisiemeReunion_0304.html">Réunion du 03/04/2018</a>
    </li>
    <li>
      <a href="quatriemeReunion_0604.html">Réunion du 06/04/2018</a>
    </li>
    <li>
      <a href="cinquiemeReunion_0405.html">Réunion du 04/05/2018</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Séminaires
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seminaires_CNRS.html">Séminaires CNRS</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-newspaper-o"></span>
     
    Biblio
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seabass.html">Dicentrarchus labax</a>
    </li>
    <li>
      <a href="sequencing_methods.html">Méthodes de séquençage</a>
    </li>
    <li>
      <a href="SNP_annotation.html">Annotation des SNPs</a>
    </li>
    <li>
      <a href="statisticSNP.html">Analyse statistiques des SNPs</a>
    </li>
    <li>
      <a href="GWAS_Stress_Fish.html">GWAS Stress Fish</a>
    </li>
    <li>
      <a href="genome_annotation.html">Annotation d'un génome</a>
    </li>
    <li>
      <a href="ManagingAffyData.html">Managing Affymetrix Data</a>
    </li>
    <li>
      <a href="pipeline_tools.html">Pipeline tools</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Annotation d’un génome</h1>

</div>


<hr />
<p>Some interesting extracts from papers :</p>
<hr />
<div id="a-beginners-guide-to-eukaryotic-genome-annotation-yandell-et-al.-2012-nature-reviews-genetics" class="section level1">
<h1>A beginner’s guide to eukaryotic genome annotation, Yandell et al., 2012, Nature Reviews Genetics</h1>
<p style="text-align:justify" ;>
Repeat identification : Identifying repeats is complicated by the fact that repeats are often poorly conserved; thus, accu- rate repeat detection usually requires users to create a repeat library for their genome of interest. Available tools for doing so generally fall into two classes: homology- based tools and de novo tools After it has been created, a repeat library can be used in conjunction with a tool such as RepeatMasker,which uses BLAST and Crossmatch34 to identify stretches of sequence in a target genome that are homologous to known repeats
</p>
<p style="text-align:justify" ;>
Evidence alignment : In prin- ciple, TBLASTX can be used to align ESTs and RNA-seq data from phylogenetically distant organisms but, owing to high computational costs, this is only done rarely. EST and protein sequence data sets are often aligned to the genome in a two-tiered process. Frequently, BLAST and BLAT are used to. identify approximate regions of homology rapidly. These alignments are usually filtered to identify and to remove marginal alignments on the basis of metrics such as percent similarity or percent identity. Clustering has two purposes. First, it groups diverse computational results into a single cluster of data, all supporting the same gene. Second, it identifies and purges redundant evidence; highly expressed genes, for example, may be supported by hundreds if not thousands of identical ESTs splice-site-aware alignment algorithms, such as Splign, Spidey , sim4 and Exonerate, are often used to realign matching and highly similar ESTs, mRNAs and proteins to the genomic input sequence. RNA-seq data have the greatest potential to improve the accuracy of gene annotations, as these data provide copious evidence for better delimitation of exons, splice sites and alter- natively spliced exons. They can be assembled de novo - that is, independently of the genome - using tools such as ABySS, SOAPdenovo and Trinity can be directly aligned to the genome using tools such as TopHat, GSNAP or Scripture followed by the assembly of alignments (rather than reads) into transcripts using tools such as Cufflinks for guidance on the best way to use TopHat with Cufflinks. Cufflinks sometimes erroneously merge RNA-seq reads from neighbouring genes. In such cases, de novo assembly of the RNA-seq data mitigates the problem; in fact, Trinity50 is designed to deal with this issue. Several annotation pipelines are now compatible with RNA-seq data: these include PASA, which uses inchworm outputs, and MAKER, which can operate directly from Cufflinks54 outputs or can use preassembled RNA-seq data.
</p>
<p style="text-align:justify" ;>
Ab initio gene prediction : most gene predictors find the single most likely coding sequence (CDS) and do not report untranslated regions (UTRs) or alternatively spliced transcripts. The MAKER pipeline provides a simplified process for training the predictors Augustus and SNAP using the EST, protein and mRNA-seq alignments that MAKER has produced. An alternative is to use GeneMark-ES: a self-training, but sometimes less-accurate, algorithm.
</p>
<p style="text-align:justify" ;>
Automated annotation : to select the single prediction whose intron-exon structure best represents the consensus of the models from among the overlapping predictions that define each putative gene locus. This is the process used by JIGSAW. EVidenceModeler (EVM) and GLEAN (and its successor, Evigan). EVM allows the user to set expected evidence error rates manually or to learn them from a training set. By contrast, GLEAN and Evigan use an unsupervised learning method to estimate a joint error model, and thus they require no additional training. In a recent gene pre- diction competition, the combiners nearly always improved on the underlying gene prediction models, and JIGSAW, EVM or Evigan performed similarly. The predictions can also be processed - before or after running the chooser - to attain still greater accuracies by having the annotation pipeline add UTRs as suggested by the RNA-seq and EST data. This is the process used by PASA, Gnomon and MAKER. the post-processed gene model that is most consistent with the protein, EST and RNA-seq alignments ; EVM, MAKER and PASA all provide methods for doing so a full run by an annotation pipeline such as MAKER or PASA can take weeks, but because these pipelines align evidence to the genome, their outputs provide starting points for annotation curation and downstream analyses, such as differential expression analyses using RNA-seq data. Another factor to consider is the phylogenetic relationship of the study genome to other annotated genomes. If it is the first of its taxonomic order or family to be annotated, it would definitely be preferable to use a pipeline that can use the full repertory of external evidence, especially RNA-seq data, to inform its gene annotations; not doing so will almost certainly result in low-quality annotations
</p>
<p style="text-align:justify" ;>
Visualizing the annotation data : Four commonly used formats for describing annotations are the GenBank, GFF3, GTF and EMBL formats. the Generic Model Organism Database (GMOD) project community has developed a series of standards and tools for description, analyses, visualization and redistribution of genome annotations, all of which use the GFF3 file format as inputs and outputs. Leveraging GMOD tools and GFF3 substantially simplifies curation, analysis, publication and management of genome annotations. Users can directly visualize these files using GBROWSE and JBROWSE to produce views of their data just like those offered at WormBase and UCSC.
</p>
<p style="text-align:justify" ;>
Quality control : Even the best gene predictors and genome annotation pipelines rarely exceed accuracies of 80% at the exon level, meaning that most gene annotations contain at least one mis-annotated exon. Given these facts, assessing how accurately a genome is annotated is an important part of any project. Assessing annotation quality in the absence of reference genome annotations is a difficult problem. Experimental verification is one solution, but few projects have the resources to carry this out on a large scale.
</p>
<p style="text-align:justify" ;>
Approaches for assessing annotation quality : One simple approach for obtaining a rough indication of annotation quality is to quantify the percentage of annotations that encode proteins with known domains using tools such as InterProScan and Pfam or tools such as MAKER, which provides an automated means for carrying out such analyses. Sequence Ontology Project has developed several metrics for quality control of genome annotations. Annotation edit distance (AED), for example, measures how congruent each annotation is with its overlapping evidence. The MAKER2 genome annotation pipeline provides some useful tools for automatically calculating AED.
</p>
<p style="text-align:justify" ;>
Submitting annotations to public databases : Ensembl and GenBank also automatically handle the heavy lifting that is involved in relating gene models to those of other organisms and identifying homologues, paralogues and orthologues. in short, they integrate a data set into the larger landscape of genomics and genome annotations. Best of all, the entire process is free, and submission to these sites in no way abridges the rights of the generators of the data to host and maintain their own genome data- base.
</p>
<p style="text-align:justify" ;>
Updating annotations : The question then becomes how to merge, update and improve the existing annotations and, at the same time, to document the process. Among existing tools, GLEAN and PASA can be used to report differences between pre-existing gene models and newly created ones. Ensembl has a proce- dure to merge annotation data sets to produce a con- sensus, and PASA has one for updating annotations with RNA-seq data. The MAKER annotation pipeline provides an automated toolkit with all of these func- tionalities and can revise, update and merge existing annotation data sets, as well as map them forwards to new assemblies. Ensembl also allows submission of such data, although the process is less formal, and interested parties should contact Ensembl directly
</p>
<p style="text-align:justify" ;>
Conclusion : Genome annotation has moved beyond merely identifying protein-coding genes to include an ever-greater emphasis on the annotation of transposons, regulatory regions, pseudogenes and ncRNA genes. Annotation quality control and management are also increasingly becoming bottlenecks. As long as tools and sequencing technologies continue to develop, periodic updates to every genome’s annotations will remain necessary. Like parenthood, annotation responsibilities do not end with birth.
</p>
</div>
<div id="improving-eukaryotic-genome-annotation-using-single-molecule-mrna-sequencing-magrini-et-al.-2018-bmc-genomics." class="section level1">
<h1>Improving eukaryotic genome annotation using single molecule mRNA sequencing, Magrini et al., 2018, BMC Genomics.</h1>
<p style="text-align:justify" ;>
PacBio sequencing is unique when compared to sequencing-by-synthesis approaches. Read lengths are proportional to reaction times (movie lengths). Thus, PacBio is becoming the gold standard in long read sequencing technologies with average polymerase reads easily exceeding 14kbp.
</p>
<p style="text-align:justify" ;>
Gene calling, annotation and comparison Repeat sequences were identified by generating a custom repeat library using Repeatmodeler (<a href="http://www.repeatmasker.org/RepeatModeler/" class="uri">http://www.repeatmasker.org/RepeatModeler/</a>). The ribosomal RNA genes were dentified using RNAmmer [37] and transfer RNAs (tRNAs) were identified using tRNAscan-SE [38]. Other non-coding RNAs (such as microRNAs) were identified by a sequence homology search of the Rfam database [39]. These repeats and predicted RNAs were thenmasked using RepeatMasker [40]. Protein-coding genes were predicted using a combination of ab initio predictors Snap [41]and Fgenesh [42] and the evidence based predictor Augustus [43]. These predictions were fed to the annotation pipeline tool Maker (version 2.26) [44] which utilizes aligned EST [20] and protein evidence, to revise the predicted gene structures.
</p>
</div>
<div id="the-state-of-play-in-higher-eukaryote-gene-annotation-mudge-et-al.-2016-nature-reviews-genetics." class="section level1">
<h1>The state of play in higher eukaryote gene annotation, Mudge et al., 2016, Nature Reviews Genetics.</h1>
<p style="text-align:justify" ;>
gene annotation is a key mechanism through which information is lever- aged from genome sequences, and deficiencies in gene- builds will be propagated into downstream analyses
</p>
<p style="text-align:justify" ;>
A key annotation question concerns the proportion of the transcriptome that con- tributes to cellular function, and it could be argued that the goal of annotation projects should be to describe only this ‘functional transcriptome’: to extract the signal from the noise
</p>
<p style="text-align:justify" ;>
We explain below how gene annotation projects are coordinating efforts to combine such data sets into fully integrated views of genomic organization.
</p>
<p style="text-align:justify" ;>
annotation defines the structure of a transcript and it provides inferences into its potential function
</p>
<p style="text-align:justify" ;>
Annotation strategies. Numerous factors come into play when choosing an annotation strategy for a genome (FIGS 2,3). Obviously, financial considera- tions can place major constraints on the availability of human resources and computational power, as well as on the generation of experimental data to provide evidence for model construction.
</p>
<p style="text-align:justify" ;>
For example, a com- mon goal of sequencing projects within evolutionary biology is to find genes that have been subjected to positive selection. In this scenario, a high premium is placed on the identification of protein-coding sequences; features such as pseudogenes and small RNAs may even be completely ignored. Meanwhile, the Functional Annotation of Animal Genomes (FAANG) consortium plans to sequence and annotate livestock genomes in order to further our understanding of quantitative phenotypes.
</p>
<p style="text-align:justify" ;>
Finally, we note that genome quality is an important factor when strat egizing. One cannot create high-quality genebuilds using poor-quality genomes, and even modest genome assembly improvements can be massively beneficial to annotation projects, as demonstrated for the honeybee.
</p>
<p style="text-align:justify" ;>
Annotation evidence. Regardless of the scientific context of an annotation project, the most important factors to influence the genebuild that is produced are the evidence and methodologies that are used for model construction.
</p>
<p style="text-align:justify" ;>
The second key source of evidence is protein sequences, but this situation is more complicated, as the field of experimental protein sequencing lags far behind that for RNA or DNA sequencing. Thus, the earliest annotation projects described coding sequences (CDSs) based on curated protein sequences from Swiss-Prot13 (TABLE 1) and through the use of ab initio ‘open read- ing frame (ORF)-finders’ (REFS 14,15). The ORF-finding strategy sought to identify CDSs through a combination of codon frequency usage and ORF size, although many translations were subsequently judged to be spurious by manual curation.
</p>
<p style="text-align:justify" ;>
Annotation workflows. The annotation ‘workflow’ chosen illustrates a second key difference between reference and non-reference genebuilds.
</p>
<p style="text-align:justify" ;>
Manual annotation is regarded as the gold standard18 , and is one of the core workflows that allows genebuilds to be classified as ‘mature’ when performed to a significant degree (FIG. 3). Nonetheless, such labour-intensive work cannot cope with the number of species genomes that are becoming available, and most new genebuilds are generated entirely in silico.
</p>
<p style="text-align:justify" ;>
Computational annotation has three core processes, depending on the resources available (FIG. 2). The first process is based on the alignment of transcript evi- dence. The second process is comparative annotation, by which the evolutionary closeness of two species allows for annotation — commonly the CDSs — to be ‘projected’ from one genome to another, or for evidence from one species to be used to build models on another. The third process is ab initio annotation, through which algorithm-based ‘gene finders’ such as GENSCAN or AUGUSTUS construct models on the basis of a priori knowledge of their likely sequence. Pure ab initio anno- tation is actually now uncommon in higher eukaryotic genomes, and these strategies are most often used in combination. The RefSeq Gnomon pipeline is a modi- fied form of GENSCAN that can perform purely ab initio annotation, although it can also integrate RNA and protein homology data when available (see The NCBI Eukaryotic Genome Annotation Pipeline). Ensembl have adapted their pipeline in a similar manner, and the less species-specific evidence that is available for a given genome, the more annotation will be based on a combi- nation of projection and ab initio modelling.
</p>
<p style="text-align:justify" ;>
AUGUSTUS remains a popular choice; although it was developed as an ab initio tool for the Human Genome Project, its modern incarnation can incorporate transcript libraries and comparative evidence, albeit with a cost in terms of speed and ease of use. For such practical reasons, researchers often annotate their genome using a simpler RNA-seq assembly pipeline such as Cufflinks.
</p>
<p style="text-align:justify" ;>
Community annotation : One solution is to manually improve the annotation in a systematic, collaborative manner based on ‘crowdsourcing. the same annotation criteria and using software such as WebApollo.
</p>
<p style="text-align:justify" ;>
Annotation in population genomics : If DNA variation is of primary interest, single nucleo- tide polymorphisms (SNPs) can simply be extracted and displayed against the main assembly for that spe- cies. <strong>Manual annota- tion can also be essential when investigating structural variants (SVs), which are of great interest to biologists owing to their association with disease and evolution.</strong>
</p>
<p style="text-align:justify" ;>
Identifying missing transcripts : Meanwhile, transcribed sequences may be absent from genebuilds for three reasons: existing models may be incomplete, that is, truncated at one or both ends; whole transcripts could be missing within existing genes; and entire genes could be absent.
</p>
<p style="text-align:justify" ;>
Meanwhile, long-read RNA-seq libraries are becoming available to improve annotation (BOX 1). It is easier to align longer reads with accuracy, although the sequen cing quality is still not comparable to that of Sanger protocols.
</p>
<p style="text-align:justify" ;>
CaptureSeq involves the use of genomic hybridization arrays to ‘pull down’ portions of the transcriptome for sequencing44. It is effective at isolating transcripts that are expressed at low levels, which may otherwise be ‘drowned out’ in whole RNA assays. CaptureSeq is typically used to identify novel genes (FIG. 4) and to target partial models for completion. The experimental set-up is laborious, however, and its use is currently limited to humans and mice.
</p>
<p style="text-align:justify" ;>
Annotating transcript end points : TESs can be identified from the 3ʹ polyadenylation tail, although there is no consistent diagnostic sequence for the TSS so it is difficult to know whether a transcript is 5ʹ truncated. Such ambiguity is problematic, because confident functional annotation depends on accurate structures. Whereas a CDS may be obvious on a full-length transcript, it could be missed on a truncated version, especially if sequencing has not encom- passed the translation initiation site (TIS) or STOP codon.
</p>
<p style="text-align:justify" ;>
Functional annotation : Traditionally, functional annotation centred on the question of which models encode protein.
</p>
<p style="text-align:justify" ;>
Distinguishing protein-coding genes and pseudogenes : Although there are a limited number of dedicated tools for the computational analysis of pseudogenes, including PseudoPipe, manual annotation remains preferable.
</p>
<p style="text-align:justify" ;>
The coding potential of alternative splicing. A second complication in CDS annotation is that protein-coding genes can make distinct proteins (isoforms) through alter- native splicing.
</p>
<p style="text-align:justify" ;>
Non-productive transcription and untranslated regions in protein-coding genes : Just as for TSSs, annotation projects generally extend models to the maximum 3ʹ distance that is supported by transcriptional evidence, and do not annotate additional models based solely on alternative TESs.
</p>
<p style="text-align:justify" ;>
Annotating proteins with experimental data : CDS annotation is interpretive because the chemistry of the protein molecule makes it far less amenable than RNA to sequencing. Ribosome profiling (RP) identifies RNA regions that are undergoing translation. Although RP has been carried out on at least six other eukary otic genomes so far — as collated by the RPFdb resource88 — these data have not yet been incorporated into the computational annotation pipelines of reference genomes.
</p>
<p style="text-align:justify" ;>
LncRNA annotation : However, lncRNA functional annotation may become more proactive: sequences such as microRNA binding sites and RNA structures are beginning to be described, and ultraviolet (UV) crosslinking immunoprecipitation followed by sequen- cing (CLIP-seq) can identify RNAs interacting with RNA-binding proteins.
</p>
<p style="text-align:justify" ;>
The prioritization of functional transcripts : Clearly, there is a close relationship between the quality of gene annotation and the accuracy of vari- ant interpretation, and yet many aspects of annotation — especially functional annotation — remain puta- tive. Putative functional annotation can introduce false positives into variant-calling workflows; for example, where LoF mutations are called in CDS exons that are not in reality coding.
</p>
<p style="text-align:justify" ;>
Conclusions : The complexity of gene annotation projects reflects the complexity that exists in eukaryotic cells, and, as we do not currently fully understand the transcrip- tome, all of our genebuilds are incomplete. Current ambiguities are most keenly felt in our own species, where nothing less than a total understanding of biol- ogy is demanded. For other projects, the ‘finish line’ may not be so far into the distance, and the length of journey taken will in many ways reflect the value of that genome to science. However, all genebuilds face challenges in how they present their resource to the public; most obviously, they must find ways to make sure that increasing complexity does not correlate with decreasing usability.
</p>
</div>

<p>Copyright &copy; 2018 | Elise GUERET</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
