<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Scripts</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Génotypage du bar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Bioinfo
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="tools_description.html">Description des outils</a>
    </li>
    <li>
      <a href="tuto.html">Tutoriaux suivis</a>
    </li>
    <li>
      <a href="Scripts.html">Scripts</a>
    </li>
    <li>
      <a href="Troubleshooting.html">Troubleshooting</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Manip'
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="LaboSemaine1.html">Semaine 1</a>
    </li>
    <li>
      <a href="LaboSemaine2.html">Semaine 2</a>
    </li>
    <li>
      <a href="LaboSemaine3.html">Semaine 3</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Réunions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="premiereReunion_2702.html">Réunion du 27/02/2018</a>
    </li>
    <li>
      <a href="deuxiemeReunion_1403.html">Réunion du 14/03/2018</a>
    </li>
    <li>
      <a href="troisiemeReunion_0304.html">Réunion du 03/04/2018</a>
    </li>
    <li>
      <a href="quatriemeReunion_0604.html">Réunion du 06/04/2018</a>
    </li>
    <li>
      <a href="cinquiemeReunion_0405.html">Réunion du 04/05/2018</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Séminaires
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seminaires_CNRS.html">Séminaires CNRS</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-newspaper-o"></span>
     
    Biblio
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seabass.html">Dicentrarchus labax</a>
    </li>
    <li>
      <a href="sequencing_methods.html">Méthodes de séquençage</a>
    </li>
    <li>
      <a href="SNP_annotation.html">Annotation des SNPs</a>
    </li>
    <li>
      <a href="statisticSNP.html">Analyse statistiques des SNPs</a>
    </li>
    <li>
      <a href="GWAS_Stress_Fish.html">GWAS Stress Fish</a>
    </li>
    <li>
      <a href="genome_annotation.html">Annotation d'un génome</a>
    </li>
    <li>
      <a href="ManagingAffyData.html">Managing Affymetrix Data</a>
    </li>
    <li>
      <a href="pipeline_tools.html">Pipeline tools</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Scripts</h1>

</div>


<hr />
<p>Quelques explications sur mes scripts :</p>
<hr />
<div id="gatk-decouverte-de-variants-a-partir-de-donnees-de-rna-seq" class="section level1">
<h1>GATK : Découverte de Variants à partir de données de RNA-seq</h1>
<p>Voici les bonnes pratiques pour faire de la découverte de variants à partir de données de RNA-seq : <img src="BPDV.png" alt="Source : https://software.broadinstitute.org/gatk/documentation/article?id=4067" /></p>
<p style="text-align:justify" ;>
Comme mentionné ci-dessus on ne peut pas filtrer les variants par la méthode de VSQR pour le moment. Cette méthode utilise du machine learning pour filtrer les données. Elle s’adapte donc aux données qu’on lui donne. Il faut donc passer par du “hard-filtering”. C’est-à-dire un filtre sur critère strict, défini comme des recommendations par l’équipe qui développe GATK. Les recommandations générales de l’équipe de GATK sont mentionnées <a href="https://software.broadinstitute.org/gatk/documentation/article.php?id=3225">ici</a>. Dans cet article ils mentionnent différents seuils pour différents type de filtres à appliquer. Ceux qu’ils proposent sont les suivants :
</p>
<table>
<thead>
<tr class="header">
<th align="center">Sigle du filtre utilisé</th>
<th align="center">Valeur si SNPs</th>
<th align="center">Valeur si INDELs</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">QD</td>
<td align="center">&lt;2.0</td>
<td align="center">&lt;2.0</td>
</tr>
<tr class="even">
<td align="center">FS</td>
<td align="center">&gt;60.0</td>
<td align="center">&gt;200.0</td>
</tr>
<tr class="odd">
<td align="center">ReadPosRankSum</td>
<td align="center">&lt;-8.0</td>
<td align="center">&lt;20.0</td>
</tr>
<tr class="even">
<td align="center">MQ</td>
<td align="center">&lt;40.0</td>
<td align="center"></td>
</tr>
<tr class="odd">
<td align="center">MQRankSum</td>
<td align="center">&lt;-12.5</td>
<td align="center"></td>
</tr>
<tr class="even">
<td align="center">SOR</td>
<td align="center">&gt;3.0</td>
<td align="center">&gt;10.0</td>
</tr>
<tr class="odd">
<td align="center">InbreedingCoeff</td>
<td align="center"></td>
<td align="center">&lt;-0.8</td>
</tr>
</tbody>
</table>
<p>Définition des filtres utilisés pour le “hard-filtering”:</p>
<ul>
<li><strong>Quality by depth (QD)</strong> :
<ul>
<li>Overview : This annotation puts the variant confidence QUAL score into perspective by normalizing for the amount of coverage available. Because each read contributes a little to the QUAL score, variants in regions with deep coverage can have artificially inflated QUAL scores, giving the impression that the call is supported by more evidence than it really is. To compensate for this, we normalize the variant confidence by depth, which gives us a more objective picture of how well supported the call is.</li>
<li>Statistical note : The QD is the QUAL score normalized by allele depth (AD) for a variant. For a single sample, the HaplotypeCaller calculates the QD by taking QUAL/AD. For multiple samples, HaplotypeCaller and GenotypeGVCFs calculate the QD by taking QUAL/AD of samples with a non hom-ref genotype call. The reason we leave out the samples with a hom-ref call is to not penalize the QUAL for the other samples with the variant call.</li>
<li>Caveats : This annotation can only be calculated for sites for which at least one sample was genotyped as carrying a variant allele.</li>
</ul></li>
<li><strong>FisherStrand bias estimated using Fischer’s exact test (FS)</strong> :
<ul>
<li>Overview : Strand bias is a type of sequencing bias in which one DNA strand is favored over the other, which can result in incorrect evaluation of the amount of evidence observed for one allele vs. the other. The FisherStrand annotation is one of several methods that aims to evaluate whether there is strand bias in the data. It uses Fisher’s Exact Test to determine if there is strand bias between forward and reverse strands for the reference or alternate allele. The output is a Phred-scaled p-value. The higher the output value, the more likely there is to be bias. More bias is indicative of false positive calls.<br />
</li>
<li>Statistical note : See the <a href="https://software.broadinstitute.org/gatk/documentation/article.php?id=8056">method document on Fischer’s exact Test</a> for a more detail explanation of this application.<br />
</li>
<li>Caveats : The FisherStrand test may not be calculated for certain complex indel cases or for multi-allelic sites. FisherStrand is best suited for low coverage situations. For testing strand bias in higher coverage situations, see the StrandOddsRatio annotation.</li>
</ul></li>
<li><strong>StrandOddsRatio (SOR)</strong> :
<ul>
<li>This is another way to estimate strand bias using a test similar to the symmetric odds ratio test. SOR was created because FS tends to penalize variants that occur at the ends of exons. Reads at the ends of exons tend to only be covered by reads in one direction and FS gives those variants a bad score. SOR will take into account the ratios of reads that cover both alleles.</li>
</ul></li>
<li><strong>RMSMappingQuality Root Mean Square of the mapping quality of reads across all samples (MQ)</strong> :
<ul>
<li>Overview : This annotation provides an estimation of the overall mapping quality of reads supporting a variant call. It produce both raw data (sum of square and num of total reads) and the calculated root mean square. The raw data is used to accurately calculate the root mean square when combining more than one sample.<br />
</li>
<li>Statistical note : The root mean square is equivalent to the mean of the mapping qualities plus the standard deviation of the mapping qualities.</li>
<li>Caveat : Uninformative reads are not used in this annotation.</li>
</ul></li>
<li><strong>MappingQualityRankSum Rank Sum Test for mapping qualities of REF versus ALT reads (MQRankSum)</strong>:
<ul>
<li>Overview : his variant-level annotation compares the mapping qualities of the reads supporting the reference allele with those supporting the alternate allele. The ideal result is a value close to zero, which indicates there is little to no difference. A negative value indicates that the reads supporting the alternate allele have lower mapping quality scores than those supporting the reference allele. Conversely, a positive value indicates that the reads supporting the alternate allele have higher mapping quality scores than those supporting the reference allele. This annotation can be used to evaluate confidence in a variant call and is a recommended covariate for variant recalibration (VQSR). Finding a statistically significant difference in quality either way suggests that the sequencing and/or mapping process may have been biased or affected by an artifact. In practice, we only filter out low negative values when evaluating variant quality because the idea is to filter out variants for which the quality of the data supporting the alternate allele is comparatively low. The reverse case, where it is the quality of data supporting the reference allele that is lower (resulting in positive ranksum scores), is not really informative for filtering variants.<br />
</li>
<li>Statistical note : The value output for this annotation is the u-based z-approximation from the Mann-Whitney-Wilcoxon Rank Sum Test for mapping qualities (MAPQ of reads supporting REF vs. MAPQ of reads supporting ALT). See the method document on statistical tests for a more detailed explanation of the ranksum test.<br />
</li>
<li>Caveat : The mapping quality rank sum test can not be calculated for sites without a mixture of reads showing both the reference and alternate alleles. AND Uninformative reads are not used in these annotations.</li>
</ul></li>
<li><strong>ReadPosRankSum Rank Sum Test for relative positioning of REF versus ALT alleles within reads (ReadPosRankSum)</strong> :
<ul>
<li>Overview : This variant-level annotation tests whether there is evidence of bias in the position of alleles within the reads that support them, between the reference and alternate alleles. Seeing an allele only near the ends of reads is indicative of error, because that is where sequencers tend to make the most errors. However, some variants located near the edges of sequenced regions will necessarily be covered by the ends of reads, so we can’t just set an absolute “minimum distance from end of read” threshold. That is why we use a rank sum test to evaluate whether there is a difference in how well the reference allele and the alternate allele are supported. The ideal result is a value close to zero, which indicates there is little to no difference in where the alleles are found relative to the ends of reads. A negative value indicates that the alternate allele is found at the ends of reads more often than the reference allele. Conversely, a positive value indicates that the reference allele is found at the ends of reads more often than the alternate allele. This annotation can be used to evaluate confidence in a variant call and is a recommended covariate for variant recalibration (VQSR). Finding a statistically significant difference in relative position either way suggests that the sequencing process may have been biased or affected by an artifact. In practice, we only filter out low negative values when evaluating variant quality because the idea is to filter out variants for which the quality of the data supporting the alternate allele is comparatively low. The reverse case, where it is the quality of data supporting the reference allele that is lower (resulting in positive ranksum scores), is not really informative for filtering variants.</li>
<li>Statistical note : The value output for this annotation is the u-based z-approximation from the Mann-Whitney-Wilcoxon Rank Sum Test for site position within reads (position within reads supporting REF vs. position within reads supporting ALT). See the <a href="https://software.broadinstitute.org/gatk/documentation/article.php?id=8031">method document on RankSum Test</a> for a more detail explanation of this test.</li>
<li>Caveat : The read position rank sum test can not be calculated for sites without a mixture of reads showing both the reference and alternate alleles. Uninformative reads are not used in these annotations.</li>
</ul></li>
</ul>
<p style="text-align:justify" ;>
Sur <a href="https://software.broadinstitute.org/gatk/documentation/article?id=6925">cette page</a>, ils présentent différents graphiques avec leur données et expliquent pourquoi ils ont choisis ces différents seuils. Ensuite il ne reste plus qu’à appliquer ces filtres aux données selon les recommendations <a href="https://software.broadinstitute.org/gatk/documentation/article.php?id=2806">suivantes</a>.
</p>
</div>
<div id="python-correction-du-fichier-dannotation-de-dicentrarchus-labrax" class="section level1">
<h1>Python : Correction du fichier d’annotation de <em>Dicentrarchus labrax</em></h1>
</div>

<p>Copyright &copy; 2018 | Elise GUERET</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
