<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Annotation fonctionnelle des SNPs</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Génotypage du bar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Bioinfo
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="tools-used.html">Outils utilisés</a>
    </li>
    <li>
      <a href="tools_description.html">Description des outils</a>
    </li>
    <li>
      <a href="tuto.html">Tutoriaux suivis</a>
    </li>
    <li>
      <a href="Troubleshooting.html">Troubleshooting</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Manip'
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="LaboSemaine1.html">Semaine 1</a>
    </li>
    <li>
      <a href="LaboSemaine2.html">Semaine 2</a>
    </li>
    <li>
      <a href="LaboSemaine3.html">Semaine 3</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Réunions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="premiereReunion_2702.html">Réunion du 27/02/2018</a>
    </li>
    <li>
      <a href="deuxiemeReunion_1403.html">Réunion du 14/03/2018</a>
    </li>
    <li>
      <a href="troisiemeReunion_0304.html">Réunion du 03/04/2018</a>
    </li>
    <li>
      <a href="quatriemeReunion_0604.html">Réunion du 06/04/2018</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Séminaires
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seminaires_CNRS.html">Séminaires CNRS</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-newspaper-o"></span>
     
    Biblio
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seabass.html">Dicentrarchus labax</a>
    </li>
    <li>
      <a href="sequencing_methods.html">Méthodes de séquençage</a>
    </li>
    <li>
      <a href="SNP_annotation.html">Annotation des SNPs</a>
    </li>
    <li>
      <a href="statisticSNP.html">Analyse statistiques des SNPs</a>
    </li>
    <li>
      <a href="GWAS_Stress_Fish.html">GWAS Stress Fish</a>
    </li>
    <li>
      <a href="genome_annotation.html">Annotation d'un génome</a>
    </li>
    <li>
      <a href="ManagingAffyData.html">Managing Affymetrix Data</a>
    </li>
    <li>
      <a href="pipeline_tools.html">Pipeline tools</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Annotation fonctionnelle des SNPs</h1>

</div>


<hr />
<p>Some interesting extracts from papers :</p>
<hr />
<div id="from-array-data" class="section level1">
<h1>From array data</h1>
<div id="david-the-database-for-annotation-visualization-and-integrated-discovery" class="section level2">
<h2>DAVID : The Database for Annotation, Visualization and Integrated Discovery</h2>
<p><a href="https://david.ncifcrf.gov/" class="uri">https://david.ncifcrf.gov/</a></p>
<p>Need a gene list and the type of array used. Not adapted for our analysis</p>
</div>
<div id="babelomics" class="section level2">
<h2>BABELOMICS</h2>
<p><a href="http://babelomics.bioinfo.cipf.es/" class="uri">http://babelomics.bioinfo.cipf.es/</a></p>
<p>a webtool</p>
</div>
<div id="fatigo-a-functional-profiling-tool-for-genomic-data.-integration-of-functional-annotation-regulatory-motifs-and-interaction-data-with-microarray-experiments-al-shahrour-et-al.-2007-nucleic-acids-research." class="section level2">
<h2>FatiGO+ : a functional profiling tool for genomic data. Integration of functional annotation, regulatory motifs and interaction data with microarray experiments, Al-Shahrour et al., 2007, Nucleic Acids Research.</h2>
<p>Cf BABELOMICS</p>
</div>
<div id="scan-snp-and-copy-number-annotation-gamazon-et-al.-2010-bioinformatics." class="section level2">
<h2>SCAN : SNP and copy number annotation, Gamazon et al., 2010, Bioinformatics.</h2>
<p>Only for Affymetrix human array</p>
</div>
</div>
<div id="from-sequencing-data" class="section level1">
<h1>From sequencing data</h1>
<div id="gatk-haplotypecaller-variantannotator" class="section level2">
<h2>GATK : HaplotypeCaller, VariantAnnotator</h2>
</div>
<div id="snpeff-a-program-for-annotating-and-predicting-the-effects-of-single-nucleotide-polymorphisms-cingolani-et-al.-2012-fly." class="section level2">
<h2>snpEff : A program for annotating and predicting the effects of single nucleotide polymorphisms, Cingolani et al., 2012, Fly.</h2>
<p>Some extracts from the paper :</p>
<p style="text-align:justify" ;>
SnpEff, which is able to analyze and annotate thousands of variants per second. SnpEff supports more genome versions, is open source for any user, supports variant call format (VCF) files and it is marginally faster. SnpEff, an abbreviation of “SNP effect,” is a multi-platform open source variant effect predictor program. SnpEff annotates variants and predicts the coding effects of genetic variations, such as SNPs, insertions and deletions (INDELs) and multiple nucle- otide polymorphisms (MNPs) (<a href="http://SnpEff.sourceforge.net/" class="uri">http://SnpEff.sourceforge.net/</a>). The main features of SnpEff include: (1) speed—the ability to make thousands of predictions per second; (2) flexibility—the ability to add custom genomes and annotations; (3) the ability to integrate with Galaxy, an open access and web-based platform for computational bioinformatic research (<a href="http://gmod.org/wiki/" class="uri">http://gmod.org/wiki/</a> Galaxy); (4) compatibility with multiple species and multiple codon usage tables (e.g., mitochondrial genomes); (5) integration with Genome Analysis Toolkit (GATK); and (6) ability to perform non-coding annotations. When SnpEff was integrated into the GATK, it replaced the ANNOVAR program for variant analyses. SnpEff algorithm : Il coupe le génome en forêt ou interval d’arbre est indexé sur le chromosome dont il provient. Chaque interval d’arbres est composé de neouds et chaque noeuds est composé de 5 éléments.
</p>
<p>-&gt; D_labrax database built. -&gt; SNP_Array annotated.</p>
</div>
<div id="vep-the-ensembl-variant-effect-predictor-mclaren-et-al.-2016-genome-biology" class="section level2">
<h2>VEP : The Ensembl Variant Effect Predictor, McLaren et al., 2016, Genome Biology</h2>
<p><a href="https://www.ensembl.org/info/docs/tools/vep/index.html" class="uri">https://www.ensembl.org/info/docs/tools/vep/index.html</a></p>
<p>webtool and command line script possible</p>
<p>Il faut avoir un bon r?seau internet pour pouvoir le t?l?charger. Ce que je ne peux pas faire pour le moment car je ne suis qu’en wi-fi et que cela ne fonctionne pas.</p>
<p>Some interesting extract from the paper :</p>
<p style="text-align:justify" ;>
A script is included in the VEP script package to create an annota- tion set from a general feature format (GFF) and FASTA file pair. The downloadable Perl script is the most powerful and flexible way to use the VEP. It supports more options than the other interfaces, has no limit on input file size, and includes extensive input, output, filtering, and analysis options. To install the script, simply download the VEP package and run the installer script, which automatically downloads the necessary API and annotation files (or ‘cache’ files). Updates with the latest data are available for each Ensembl release. The full source code is freely available on the Ensembl GitHub repository. To process large volumes of data, the VEP script works most efficiently in “offline” mode using a local cache of transcript annotations rather than online public databases. As well as optimizing runtime, this ensures data privacy for clinically or commercially sensitive data. Furthermore, the VEP input can be configured to query overlaps with local, potentially private, variant and phenotype data or other custom data sets in a manner similar to vcfanno. In this way annotation in for- mats including BED, GFF, GTF, VCF, and bigWig can be incorporated into the VEP output.
</p>
</div>
<div id="variant-tools" class="section level2">
<h2>variant tools</h2>
<p><a href="http://varianttools.sourceforge.net/" class="uri">http://varianttools.sourceforge.net/</a></p>
<p>Tutorials : <a href="http://varianttools.sourceforge.net/Main/Documentation" class="uri">http://varianttools.sourceforge.net/Main/Documentation</a></p>
</div>
<div id="blast2go" class="section level2">
<h2>Blast2Go</h2>
<p><a href="https://www.blast2go.com/" class="uri">https://www.blast2go.com/</a></p>
<p>purchase licence to use it</p>
</div>
<div id="fastsnp-an-always-up-to-date-and-extendable-service-for-snp-function-analysis-and-prioritization-yuan-et-al.-2006-nucleic-acids-research." class="section level2">
<h2>FASTSNP : An always up-to-date and extendable service for SNP function analysis and prioritization, Yuan et al., 2006, Nucleic Acids Research.</h2>
<p><a href="https://omictools.com/function-analysis-and-selection-tool-for-single-nucleotide-polymorphisms-tool" class="uri">https://omictools.com/function-analysis-and-selection-tool-for-single-nucleotide-polymorphisms-tool</a></p>
<p>Not available anymore Only for human data</p>
</div>
<div id="annovar-genomic-variant-annotation-and-prioritization-with-annovar-and-wannovar-yang-et-al.-2015-nature-protocols." class="section level2">
<h2>ANNOVAR : Genomic variant annotation and prioritization with ANNOVAR and wANNOVAR, Yang et al., 2015, Nature Protocols.</h2>
<p>only for human data but can be adaptated to human maybe</p>
</div>
<div id="snpdat-easy-and-rapid-annotation-of-results-from-de-novo-snp-discovery-projects-for-model-and-non-model-organisms-doran-et-al.-2013-bmc-bioinformatics" class="section level2">
<h2>Snpdat: Easy and rapid annotation of results from de novo snp discovery projects for model and non-model organisms, doran et al., 2013, BMC Bioinformatics</h2>
<p>Annotation de novo sur des organismes non-model</p>
</div>
<div id="vai-from-ucsc" class="section level2">
<h2>VAI from UCSC</h2>
<p>webtool work only with its database</p>
</div>
<div id="seattleseq-annotation" class="section level2">
<h2>SeattleSeq annotation</h2>
<p>human hg 19 only</p>
</div>
<div id="variant-variant-analysis-tool" class="section level2">
<h2>Variant : VARIant ANalysis Tool</h2>
<p>On human mouse and rat</p>
</div>
<div id="vat" class="section level2">
<h2>VAT</h2>
<p>On human with many depedencies to download.</p>
</div>
<div id="funseq2" class="section level2">
<h2>Funseq2</h2>
<p>Regulatory mutation from cancer genome sequencing =&gt; human</p>
</div>
</div>

<p>Copyright &copy; 2018 | Elise GUERET</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
