<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Analyse statistiques des SNPs</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Génotypage du bar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Bioinfo
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="tools-used.html">Outils utilisés</a>
    </li>
    <li>
      <a href="tools_description.html">Description des outils</a>
    </li>
    <li>
      <a href="tuto.html">Tutoriaux suivis</a>
    </li>
    <li>
      <a href="Troubleshooting.html">Troubleshooting</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Manip'
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="LaboSemaine1.html">Semaine 1</a>
    </li>
    <li>
      <a href="LaboSemaine2.html">Semaine 2</a>
    </li>
    <li>
      <a href="LaboSemaine3.html">Semaine 3</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Réunions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="premiereReunion_2702.html">Réunion du 27/02/2018</a>
    </li>
    <li>
      <a href="deuxiemeReunion_1403.html">Réunion du 14/03/2018</a>
    </li>
    <li>
      <a href="troisiemeReunion_0304.html">Réunion du 03/04/2018</a>
    </li>
    <li>
      <a href="quatriemeReunion_0604.html">Réunion du 06/04/2018</a>
    </li>
    <li>
      <a href="cinquiemeReunion_0405.html">Réunion du 04/05/2018</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Séminaires
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seminaires_CNRS.html">Séminaires CNRS</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-newspaper-o"></span>
     
    Biblio
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seabass.html">Dicentrarchus labax</a>
    </li>
    <li>
      <a href="sequencing_methods.html">Méthodes de séquençage</a>
    </li>
    <li>
      <a href="SNP_annotation.html">Annotation des SNPs</a>
    </li>
    <li>
      <a href="statisticSNP.html">Analyse statistiques des SNPs</a>
    </li>
    <li>
      <a href="GWAS_Stress_Fish.html">GWAS Stress Fish</a>
    </li>
    <li>
      <a href="genome_annotation.html">Annotation d'un génome</a>
    </li>
    <li>
      <a href="ManagingAffyData.html">Managing Affymetrix Data</a>
    </li>
    <li>
      <a href="pipeline_tools.html">Pipeline tools</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Analyse statistiques des SNPs</h1>

</div>


<hr />
<p>Some interesting extracts from papers :</p>
<hr />
<div id="comparing-methods-for-detecting-multilocus-adaptation-with-multivariate-genotype-environment-associations-forester-et-al.-2017-molecular-ecology" class="section level1">
<h1>Comparing methods for detecting multilocus adaptation with multivariate genotype-environment associations, Forester et al., 2017, Molecular Ecology</h1>
<p>We found that constrained ordinations, in particular redundancy analysis (RDA), show a superior combination of low FPRs and high TPRs across weak, moderate and strong multilocus selection. These results were robust across the levels of population structure, demo- graphic histories, sampling designs and sample sizes tested here. Additionally, RDA outperformed an alternative ordination-based approach, cRDA, especially (and importantly) when the multilocus selection signature was completely derived from loci under weak selection. It is important to note that population structure was rela- tively low in these simulations. Results may differ for systems with strong population structure or metapopulation dynamics, where it can be important to correct for structure or combine detections with another GEA that accounts for structure. Continued testing of these promising methods is needed in simulation frameworks that include more population structure, multiple selection surfaces and genetic architectures that are more complex than the multilocus selection response modelled here. However, this study indicates that con- strained ordinations are an effective means of detecting adaptive processes that result in weak, multilocus molecular signatures, pro- viding a powerful tool for investigating the genetic basis of local adaptation and informing management actions to conserve the evo- lutionary potential of species of agricultural, forestry, fisheries and conservation consern.</p>
</div>
<div id="plink-a-tool-set-for-whole-genome-association-and-population-based-linkage-analyses-purcell-etal.-2007-the-american-journal-of-human-genetics" class="section level1">
<h1>PLINK : A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses, Purcell etal., 2007, The American Journal of Human Genetics</h1>
<p>whole-genome association studies (WGAS). A typical WGAS, currently with hundreds of thousands of SNPs ge- notyped for thousands of individuals, represents a data set that is several orders of magnitude larger than previous linkage and association studies. For example, if 500 individuals are needed to test a single SNP with adequate power, ∼2,000 individuals will be required to test 500,000 SNPs, even after Bonferroni correction When a relatively small WGAS data set of 100,000 SNPs genotyped for 350 individuals is considered, for example, PLINK takes ∼10 s to load, filter, and perform association analysis for all SNPs; straightforward handling of much larger data sets is also possible. Because normal screening procedures based on measures such as overall genotyping rate and Hardy-Weinberg equi- librium will often not detect these biased SNPs, it is im- portant to look closely at patterns of genotyping failure for nonrandom effects (as well as to visually inspect the raw data before calling genotypes). In PLINK, genotyping failure can be examined with respect to both phenotype and (potentially unobserved) genotype. In summary, given the issues raised above, we designed the PLINK WGAS tool set to meet the following require- ments: (a) to provide a simple way to handle large WGAS data sets, (b) to assess confounding due to stratification and nonrandom genotyping failure and to produce a range of other summary statistics, (c) to perform a variety of standard association tests efficiently on very large data sets (in populations or families, for disease or quantitative outcomes, allowing for covariates, haplotypic tests, etc.), and (d) to provide a means of assaying rare variation with the use of common SNP panels, thereby providing amap- ping method that might perform better when the MRV model holds. Data management. Summary statistics.—Standard summary measures are available: genotyping rates, allele and genotype frequen- cies, Hardy-Weinberg equilibrium tests using asymptotic and exact procedures, and single-SNP Mendelian error summaries for family data. PLINK also estimates individ- ual heterozygosity rates. Population stratification.—On the basis of the genome- wide average proportion of alleles shared identical by state (IBS) between any two individuals, PLINK offers tools to (a) cluster individuals into homogeneous subsets, (b) per- form classical multidimensional scaling (MDS) to visualize substructure and provide quantitative indices of popula- tion genetic variation, and (c) identify outlying individ- uals. PLINK uses complete-linkage hierarchical clustering to assess population stratification, with the use of whole- genome SNP data. PLINK also provides an alternate way to look at population stratification: rather than clustering into discrete groups, it can use the data-reduction technique of classical MDS to produce a k-dimensional representation of any substructure. PLINK also supports an IBS-based “nearest- neighbor” analysis to detect outlying individuals who do not belong with any major cluster in the sample. For each individual, the distance to its nearest neighbor is calcu- lated; this distribution is standardized (by the sample mean and variance of nearest-neighbor distances) and can be inspected for outliers. The same procedure can also be applied to individuals’ nth-nearest neighbor. Association analysis.—As well as the standard case/con- trol allelic test, PLINK offers a Cochran-Armitage trend test, Fisher’s exact test, genotypic tests (general,dominant, and recessive models), and Cochran-Mantel-Haenszeltests for stratified tables, which allow for tests of association conditional on any cluster solution or other categorization of samples. The Breslow-Day and homogeneity of odds ratio tests are supported, as are tests for quantitative traits that use a standard linear regression of phenotypeonallele dosage. For family data, the standard transmission/disequilib- rium test (TDT) is provided. The permutation procedure applied to the TDT flips the transmitted and untransmit- ted alleles of all individuals in a nuclear family for all SNPs per permutation, thereby preserving in each permuted data set the possible nonindependence of transmissions across SNPs and across multiple offspring due to LD and linkage A range of multiple-test corrections are also available, including those based on Bonferroni correction and false-discovery rate. IBD estimation : One requirement of this approach is that SNPs are in approximate linkage equilibrium in the population; oth- erwise, many small regions of high LD will be called as shared IBD segments. One approach is to prune the SNP panel to a reduced subset of approximately independent SNPs. Performance.—Very large WGAS data sets can be ana- lyzed using fairly standard hardware, and there are no fixed limits on the number of samples or SNPs. Graphical user interface.—We have also developed a sep- arate, optional Java-based graphical user interface, gPLINK, to initiate, track, and record PLINK jobs</p>
</div>
<div id="shesis-a-powerful-software-platform-for-analyses-of-linkage-disequilibrium-haplotype-construction-and-genetic-association-at-polymorphism-loci-she-et-al.-2005-cell-research" class="section level1">
<h1>SHEsis : a powerful software platform for analyses of linkage disequilibrium, haplotype construction, and genetic association at polymorphism loci, she et al., 2005, Cell research</h1>
</div>
<div id="mapchart-software-for-the-graphical-presentation-of-linkage-maps-and-qtls-voorips-et-al.-2002-journal-of-heredity" class="section level1">
<h1>MapChart : Software for the Graphical Presentation of Linkage Maps and QTLs, Voorips et al., 2002, Journal of heredity</h1>
</div>
<div id="gcta-a-tool-for-genome-wide-complex-trait-analysis-yang-et-al.-2011-the-american-journal-of-human-genetics" class="section level1">
<h1>GCTA : A tool for genome-wide complex trait analysis, Yang et al., 2011, The American Journal of Human Genetics</h1>
</div>
<div id="r-packages" class="section level1">
<h1>R-Packages</h1>
<div id="gwastools" class="section level2">
<h2>GWASTools</h2>
<p><a href="http://bioconductor.org/packages/release/bioc/html/GWASTools.html" class="uri">http://bioconductor.org/packages/release/bioc/html/GWASTools.html</a></p>
</div>
<div id="adegenet" class="section level2">
<h2>adegenet</h2>
</div>
<div id="genetics" class="section level2">
<h2>genetics</h2>
</div>
<div id="haplo.stats" class="section level2">
<h2>haplo.stats</h2>
</div>
<div id="gap" class="section level2">
<h2>gap</h2>
</div>
<div id="asmap" class="section level2">
<h2>ASMap</h2>
</div>
<div id="identity" class="section level2">
<h2>identity</h2>
</div>
<div id="pedigree" class="section level2">
<h2>pedigree</h2>
</div>
<div id="hwde" class="section level2">
<h2>hwde</h2>
</div>
<div id="multic" class="section level2">
<h2>multic</h2>
</div>
<div id="ldheatmap" class="section level2">
<h2>LDheatmap</h2>
</div>
<div id="snpassoc" class="section level2">
<h2>SNPassoc</h2>
</div>
<div id="snpmatrix" class="section level2">
<h2>snpMatrix</h2>
</div>
<div id="genabel" class="section level2">
<h2>GenABEL</h2>
</div>
<div id="snpstats" class="section level2">
<h2>SNPStats</h2>
</div>
</div>

<p>Copyright &copy; 2018 | Elise GUERET</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
