<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Analyse statistiques des SNPs</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Génotypage du bar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Bioinfo
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="tools_description.html">Description des outils</a>
    </li>
    <li>
      <a href="tuto.html">Tutoriaux suivis</a>
    </li>
    <li>
      <a href="Scripts.html">Scripts</a>
    </li>
    <li>
      <a href="Troubleshooting.html">Troubleshooting</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Manip'
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="LaboSemaine1.html">Semaine 1</a>
    </li>
    <li>
      <a href="LaboSemaine2.html">Semaine 2</a>
    </li>
    <li>
      <a href="LaboSemaine3.html">Semaine 3</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Réunions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="premiereReunion_2702.html">Réunion du 27/02/2018</a>
    </li>
    <li>
      <a href="deuxiemeReunion_1403.html">Réunion du 14/03/2018</a>
    </li>
    <li>
      <a href="troisiemeReunion_0304.html">Réunion du 03/04/2018</a>
    </li>
    <li>
      <a href="quatriemeReunion_0604.html">Réunion du 06/04/2018</a>
    </li>
    <li>
      <a href="cinquiemeReunion_0405.html">Réunion du 04/05/2018</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Séminaires
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seminaires_CNRS.html">Séminaires CNRS</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-newspaper-o"></span>
     
    Biblio
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seabass.html">Dicentrarchus labax</a>
    </li>
    <li>
      <a href="sequencing_methods.html">Méthodes de séquençage</a>
    </li>
    <li>
      <a href="SNP_annotation.html">Annotation des SNPs</a>
    </li>
    <li>
      <a href="statisticSNP.html">Analyse statistiques des SNPs</a>
    </li>
    <li>
      <a href="GWAS_Stress_Fish.html">GWAS Stress Fish</a>
    </li>
    <li>
      <a href="genome_annotation.html">Annotation d'un génome</a>
    </li>
    <li>
      <a href="ManagingAffyData.html">Managing Affymetrix Data</a>
    </li>
    <li>
      <a href="pipeline_tools.html">Pipeline tools</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Analyse statistiques des SNPs</h1>

</div>


<hr />
<p>Some interesting extracts from papers :</p>
<hr />
<div id="comparing-methods-for-detecting-multilocus-adaptation-with-multivariate-genotype-environment-associations-forester-et-al.-2017-molecular-ecology" class="section level1">
<h1>Comparing methods for detecting multilocus adaptation with multivariate genotype-environment associations, Forester et al., 2017, Molecular Ecology</h1>
<p style="text-align:justify" ;>
We found that constrained ordinations, in particular redundancy analysis (RDA), show a superior combination of low FPRs and high TPRs across weak, moderate and strong multilocus selection. These results were robust across the levels of population structure, demo- graphic histories, sampling designs and sample sizes tested here. Additionally, RDA outperformed an alternative ordination-based approach, cRDA, especially (and importantly) when the multilocus selection signature was completely derived from loci under weak selection. It is important to note that population structure was rela- tively low in these simulations. Results may differ for systems with strong population structure or metapopulation dynamics, where it can be important to correct for structure or combine detections with another GEA that accounts for structure. Continued testing of these promising methods is needed in simulation frameworks that include more population structure, multiple selection surfaces and genetic architectures that are more complex than the multilocus selection response modelled here. However, this study indicates that con- strained ordinations are an effective means of detecting adaptive processes that result in weak, multilocus molecular signatures, pro- viding a powerful tool for investigating the genetic basis of local adaptation and informing management actions to conserve the evo- lutionary potential of species of agricultural, forestry, fisheries and conservation consern.
</p>
</div>
<div id="plink-a-tool-set-for-whole-genome-association-and-population-based-linkage-analyses-purcell-etal.-2007-the-american-journal-of-human-genetics" class="section level1">
<h1>PLINK : A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses, Purcell etal., 2007, The American Journal of Human Genetics</h1>
<p style="text-align:justify" ;>
whole-genome association studies (WGAS).
</p>
<p style="text-align:justify" ;>
A typical WGAS, currently with hundreds of thousands of SNPs ge- notyped for thousands of individuals, represents a data set that is several orders of magnitude larger than previous linkage and association studies. For example, if 500 individuals are needed to test a single SNP with adequate power, ∼2,000 individuals will be required to test 500,000 SNPs, even after Bonferroni correction. When a relatively small WGAS data set of 100,000 SNPs genotyped for 350 individuals is considered, for example, PLINK takes ∼10 s to load, filter, and perform association analysis for all SNPs; straightforward handling of much larger data sets is also possible. Because normal screening procedures based on measures such as overall genotyping rate and Hardy-Weinberg equi- librium will often not detect these biased SNPs, it is im- portant to look closely at patterns of genotyping failure for nonrandom effects (as well as to visually inspect the raw data before calling genotypes). In PLINK, genotyping failure can be examined with respect to both phenotype and (potentially unobserved) genotype. In summary, given the issues raised above, we designed the PLINK WGAS tool set to meet the following require- ments: (a) to provide a simple way to handle large WGAS data sets, (b) to assess confounding due to stratification and nonrandom genotyping failure and to produce a range of other summary statistics, (c) to perform a variety of standard association tests efficiently on very large data sets (in populations or families, for disease or quantitative outcomes, allowing for covariates, haplotypic tests, etc.), and (d) to provide a means of assaying rare variation with the use of common SNP panels, thereby providing amap- ping method that might perform better when the MRV model holds.
</p>
<p style="text-align:justify" ;>
Data management.
</p>
<p style="text-align:justify" ;>
Summary statistics.—Standard summary measures are available: genotyping rates, allele and genotype frequen- cies, Hardy-Weinberg equilibrium tests using asymptotic and exact procedures, and single-SNP Mendelian error summaries for family data. PLINK also estimates individ- ual heterozygosity rates.
</p>
<p style="text-align:justify" ;>
Population stratification.—On the basis of the genome- wide average proportion of alleles shared identical by state (IBS) between any two individuals, PLINK offers tools to (a) cluster individuals into homogeneous subsets, (b) per- form classical multidimensional scaling (MDS) to visualize substructure and provide quantitative indices of popula- tion genetic variation, and (c) identify outlying individ- uals. PLINK uses complete-linkage hierarchical clustering to assess population stratification, with the use of whole- genome SNP data. PLINK also provides an alternate way to look at population stratification: rather than clustering into discrete groups, it can use the data-reduction technique of classical MDS to produce a k-dimensional representation of any substructure. PLINK also supports an IBS-based “nearest- neighbor” analysis to detect outlying individuals who do not belong with any major cluster in the sample. For each individual, the distance to its nearest neighbor is calcu- lated; this distribution is standardized (by the sample mean and variance of nearest-neighbor distances) and can be inspected for outliers. The same procedure can also be applied to individuals’ nth-nearest neighbor.
</p>
<p style="text-align:justify" ;>
Association analysis.—As well as the standard case/con- trol allelic test, PLINK offers a Cochran-Armitage trend test, Fisher’s exact test, genotypic tests (general,dominant, and recessive models), and Cochran-Mantel-Haenszeltests for stratified tables, which allow for tests of association conditional on any cluster solution or other categorization of samples. The Breslow-Day and homogeneity of odds ratio tests are supported, as are tests for quantitative traits that use a standard linear regression of phenotypeonallele dosage. For family data, the standard transmission/disequilib- rium test (TDT) is provided. The permutation procedure applied to the TDT flips the transmitted and untransmit- ted alleles of all individuals in a nuclear family for all SNPs per permutation, thereby preserving in each permuted data set the possible nonindependence of transmissions across SNPs and across multiple offspring due to LD and linkage A range of multiple-test corrections are also available, including those based on Bonferroni correction and false-discovery rate. IBD estimation : One requirement of this approach is that SNPs are in approximate linkage equilibrium in the population; oth- erwise, many small regions of high LD will be called as shared IBD segments. One approach is to prune the SNP panel to a reduced subset of approximately independent SNPs.
</p>
<p style="text-align:justify" ;>
Performance.—Very large WGAS data sets can be analyzed using fairly standard hardware, and there are no fixed limits on the number of samples or SNPs.
</p>
<p style="text-align:justify" ;>
Graphical user interface.—We have also developed a separate, optional Java-based graphical user interface, gPLINK, to initiate, track, and record PLINK jobs.
</p>
</div>
<div id="shesis-a-powerful-software-platform-for-analyses-of-linkage-disequilibrium-haplotype-construction-and-genetic-association-at-polymorphism-loci-she-et-al.-2005-cell-research" class="section level1">
<h1>SHEsis : a powerful software platform for analyses of linkage disequilibrium, haplotype construction, and genetic association at polymorphism loci, she et al., 2005, Cell research</h1>
<p style="text-align:justify" ;>
ABSTRACT : In multiloci-based genetic association studies of complex diseases, a powerful and high efficient tool for analyses of linkage disequilibrium (LD) between markers, haplotype distributions and many chi-square/p values with a large number of samples has been sought for long. In order to achieve the goal of obtaining meaningful results directly from raw data, we developed a robust and user-friendly software platform with a series of tools for analysis in association study with high efficiency. The platform has been well evaluated by several sets of real data.
</p>
<p style="text-align:justify" ;>
In order to detect the reliability of our platform, we evaluated it with several published scientific papers[2-6] and obtained exactly the same result but very robust rate. Compared with the tools used in cited papers[2-6], our platform shows the advantage of easier handling and the capability of analyzing complicated data. Therefore, it could become a powerful and useful tool for studies of complex diseases in the near future. The SHEsis software platform is now available for free at the above website address. And we also will go on updating its power and capacity.
</p>
</div>
<div id="mapchart-software-for-the-graphical-presentation-of-linkage-maps-and-qtls-voorips-et-al.-2002-journal-of-heredity" class="section level1">
<h1>MapChart : Software for the Graphical Presentation of Linkage Maps and QTLs, Voorips et al., 2002, Journal of heredity</h1>
<p style="text-align:justify" ;>
These charts can easily be exported to word processing or presentation software. The charts are exported as vector graphics (Windows enhanced metafiles) rather than bitmaps, which makes them easy to rescale and to edit further if desired.
</p>
<p style="text-align:justify" ;>
MapChart has a standard Windows inter- face, with a main menu, toolbar, and pop- up menus.
</p>
<p style="text-align:justify" ;>
MapChart imports linkage data from map files, produced by software for linkage analysis. Map files are composed of a se- quence of linkage groups, each with a header line specifying the linkage group ti- tle, followed by a sequence of lines with locus names and map positions.
</p>
</div>
<div id="gcta-a-tool-for-genome-wide-complex-trait-analysis-yang-et-al.-2011-the-american-journal-of-human-genetics" class="section level1">
<h1>GCTA : A tool for genome-wide complex trait analysis, Yang et al., 2011, The American Journal of Human Genetics</h1>
<p style="text-align:justify" ;>
ABSTRACT : For most human complex diseases and traits, SNPs identified by genome-wide association studies (GWAS) explain only a small fraction of the heritability. Here we report a user-friendly software tool called genome-wide complex trait analysis (GCTA), which was developed based on a method we recently developed to address the ‘‘missing heritability’’ problem. GCTA estimates the variance explained by all the SNPs on a chromosome or on the whole genome for a complex trait rather than testing the association of any particular SNP to the trait. We introduce GCTA’s five main functions: data management, estimation of the genetic relationships from SNPs, mixed linear model analysis of variance explained by the SNPs, estimation of the linkage disequilibrium structure, and GWAS simulation. The GCTA software is a versatile tool to estimate and partition complex trait variation with large GWAS data sets.
</p>
<p style="text-align:justify" ;>
There has not been any consensus on the explanation of the ‘‘missing heritability.’’ Possible explanations include a large number of common variants with small effects, rare variants with large effects, and DNA structural variation. Here we report a versatile tool called genome-wide complex trait analysis (GCTA), which implements the method of estimating variance explained by all SNPs, and extend the method to partition the genetic variance onto each of the chromosomes and also to esti- mate the variance explained by the X chromosome and test for dosage compensation in females. We developed GCTAin five function domains: data management, estima- tion of the GRM from a set of SNPs, estimation of the vari- ance explained by all the SNPs on a single chromosome or the whole genome, estimation of linkage disequilibrium (LD) structure, and simulation.
</p>
<p style="text-align:justify" ;>
Estimation of the Genetic Relationship from Genome-wide SNPs One of the core functions of GCTA is to estimate the genetic relationships between individuals from the SNPs. As a by-product, we provide a function in GCTA to calculate the eigenvectors of the GRM, which is asymptot- ically equivalent to those from the PCA implemented in EIGENSTRAT11 because the GRM (Ajk) defined in GCTA is approximately half of the covariance matrix (Jjk) used in EIGENSTRAT. The only purpose of developing this func- tion is to calculate eigenvectors and then include them in the model as covariates to capture variance due to population structure. More sophisticated analyses of the population structure can be found in programs such as EIGENSTRAT11 and STRUCTURE.
</p>
<p style="text-align:justify" ;>
In GCTA, we provide flexible options to specify different genetic models : To estimate the variance of genotype-environment interaction effects. <strong>From the REML analysis, GCTA has an option to provide the best linear unbiased prediction (BLUP) of the total genetic effect for all individuals. BLUP is widely used by plant and animal breeders to quantify the breeding value of individuals in artificial selection programs14 and also by evolutionary geneticists.</strong> <strong>For example, GCTA could be used to predict SNP effects in a discovery set, and the SNP effects could be used in PLINK to predict whole-genome profiles via the scoring approach in a validation set.</strong>
</p>
<p style="text-align:justify" ;>
In a standard GWAS, particularly with a large sample size, the mean (lmean) or median (lmedian) of the test statistics for single-SNP associations often deviates from its expected value under the null hypothesis of no association between any SNP and the phenotype, which is usually interpreted as the effect due to population stratification and/or cryptic relatedness. Given a causal variant, we use simple regression to test for SNPs in LD with the causal variant within dMbdistance in either direction. <strong>PLINK has an option (‘‘show targets’’) to select SNPs in LDwith a set of target SNPs with LD r2 larger than a user-specified cutoff value. This function is very useful to distinguish indepen- dent association signals but less suited to predict lmean and lmedian, because the test statistics of the SNPs in modest LD with causal variants (SNPs at Mb distance with low r2) will also be inflated to a certain extent, and these test statistics will contribute to the genomic inflation factors.</strong>
</p>
<p style="text-align:justify" ;>
We chose the PLINK25 compact binary file format (.bed, .bim,and .fam) as the input data format forGCTAbecause of its popularity in the genetics community and its effi- ciency of data storage. However,wedo not provide functions for a thorough quality control (QC) of the data, such as Hardy-Weinberg equilibrium test and missingness, because these functions have been well developed in many other genetic analysis packages, e.g., PLINK. <strong>We assume that the data have been cleaned by a standard QC process before entering into GCTA.</strong> When the sample size is small, e.g., n &lt; 3000, both GCTA and ASREML take a few minutes to run. When the sample size is large, e.g., n&gt;10,000, especially when fitting multiple GRMs, it takes days for ASREML to finish the analysis, whereas GCTA needs only a few hours. The estimate of variance captured by all of the SNPs obtained in GCTA is directly comparable to the heritability estimated from pedigree analysis in family and twin studies, as well as the variance explained by GWAS hits, so that missing and hiding heritability can be quantified.5 Other sources of genetic variations such as dominance, gene-gene interaction, and gene-environment interaction are also important for complex trait variation but are less relevant to the ‘‘missing heritability’’ problem if the total heritability refers to the narrow-sense heritability, i.e., the proportion of phenotypic variance due to additive genetic variance. The current version of GCTA only provides functions to estimate and partition the variances of additive and additive-environment interaction effects. It is technically feasible to extend the analysis to include dominance and/or gene-gene interaction effects in the future. Future developments will also include options to do multivariate analyses, to read genotype or imputed probability data in different formats, and to implement other applications of whole-genome or chromosome segment approaches.
</p>
<p style="text-align:justify" ;>
<strong>GCTA is not limited to the analysis of data on human complex traits, but in this report we only use examples and specifications (e.g., the number of autosomes) for humans.</strong>
</p>
</div>
<div id="genetix-a-windows-software-belkhir-et-al.-1998-2005" class="section level1">
<h1>GENETIX a windows software, Belkhir et al., 1998-2005</h1>
<p><a href="http://kimura.univ-montp2.fr/genetix/">Genetix</a> est un logiciel qui fonctionne sous Windows et qui permet de calculer un certain nombre de paramètres utilisés en Génétique des populations. Il propose également de tester leur significativité en réalisant des test de ré-échantillonnage par permutations.</p>
</div>
<div id="r-packages" class="section level1">
<h1>R-Packages</h1>
<div id="gwastools" class="section level2">
<h2>GWASTools</h2>
<p><a href="http://bioconductor.org/packages/release/bioc/html/GWASTools.html" class="uri">http://bioconductor.org/packages/release/bioc/html/GWASTools.html</a></p>
</div>
<div id="vegan" class="section level2">
<h2>vegan</h2>
<p>Mentionned by Forester 2017. Ce package permet de réaliser des analyses de rendondance.</p>
</div>
<div id="adegenet" class="section level2">
<h2>adegenet</h2>
</div>
<div id="genetics" class="section level2">
<h2>genetics</h2>
</div>
<div id="haplo.stats" class="section level2">
<h2>haplo.stats</h2>
</div>
<div id="gap" class="section level2">
<h2>gap</h2>
</div>
<div id="asmap" class="section level2">
<h2>ASMap</h2>
</div>
<div id="identity" class="section level2">
<h2>identity</h2>
</div>
<div id="pedigree" class="section level2">
<h2>pedigree</h2>
</div>
<div id="hwde" class="section level2">
<h2>hwde</h2>
</div>
<div id="multic" class="section level2">
<h2>multic</h2>
</div>
<div id="ldheatmap" class="section level2">
<h2>LDheatmap</h2>
</div>
<div id="snpassoc" class="section level2">
<h2>SNPassoc</h2>
</div>
<div id="snpmatrix" class="section level2">
<h2>snpMatrix</h2>
</div>
<div id="genabel" class="section level2">
<h2>GenABEL</h2>
</div>
<div id="snpstats" class="section level2">
<h2>SNPStats</h2>
</div>
</div>

<p>Copyright &copy; 2018 | Elise GUERET</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
